packages/http-client-csharp/generator/Microsoft.Generator.CSharp/src/Providers/ModelProvider.cs

```diff
@@ -40,17 +48,24 @@ public ModelProvider(InputModelType inputModel)
                 _declarationModifiers |= TypeSignatureModifiers.Abstract;
             }
 
+            if (inputModel.BaseModel is not null)
+            {
+                _baseTypeProvider = new(() => CodeModelPlugin.Instance.TypeFactory.CreateModel(inputModel.BaseModel));
+            }
+
             _isStruct = inputModel.ModelAsStruct;
         }
 
+        private ModelProvider? BaseModelProvider
+            => _baseModelProvider ??= (_baseTypeProvider?.Value is ModelProvider baseModelProvider ? baseModelProvider : null);
+        private ConstructorProvider SecondaryConstructor => _secondaryConstructor ??= BuildSecondaryConstructor();
```

Comment:
instead of SecondaryConstructor lets call it FullConstructor to indicate its purpose is to provide a way to set all properties